appx: Update Publisher, PublisherDisplayName, Version and Identity Name

jprvita opened the pull request:
Some of these values have restrictions or need to match what is set in our Microsoft Developer account.

https://phabricator.endlessm.com/T33456

jprvita committed:
appx: Fix publisher display name

This needs to match our display name in the Microsoft Store, "Endless OS
Foundation".

https://phabricator.endlessm.com/T33456

jprvita committed:
appx: Bump version to 1.0.0.0

When trying to submit a test package to the Microsoft store I got the
following error:

 Package acceptance validation error: Apps are not allowed to have a
 Version with a revision number other than zero specified in the app
 manifest. The package endlesskeyapp.appx specifies 0.1.10.1.

According to [1], the version format is "Major.Minor.Build.Revision",
where "Major" can never be 0.

This commit bumps the version to 1.0.0.0 to satisfy both requirements.

[1] https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-identity

https://phabricator.endlessm.com/T33456

jprvita committed:
appx: Update Identity Name

"It's redundant to call an app an app" (wjt, 2022).

https://phabricator.endlessm.com/T33456

wjt commented on .buildkite_win/appxmanifest.xml:
```diff
@@ -4,10 +4,10 @@
   xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
   xmlns:uap3="http://schemas.microsoft.com/appx/manifest/uap/windows10/3"
   xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10">
-  <Identity Name="EndlessOSFoundation.EndlessKey.app" Version="0.1.10.1" Publisher="CN=EOSF Microsoft Publishing 2022, O=Endless OS Foundation LLC" ProcessorArchitecture="x64" />
```
Let's linewrap this so each property is on its own line. We are about to have 3 PRs changing different properties on this line:

- this one
- #44 changing Publisher
- another changing ProcessorArchitecture

jprvita replied:
I've created https://github.com/endlessm/kolibri-app/pull/45 to address that, we can rebase the other PRs once that one gets merged.

jprvita committed:
appx: Adjust publisher to our Windows Publisher ID

The Microsoft Store expects the publisher field in the identity tag to
be CN=<Windows Publishing ID>. Otherwise we get the following error when
uploading an app for submission:

 Invalid package publisher name: CN=EOSF Microsoft Publishing 2022,
 O=Endless OS Foundation LLC (expected: CN=4B001A25-B854-44D9-BAD3-8AD7FC7A8761)

https://phabricator.endlessm.com/T33456

wjt commented on .buildkite_win/appxmanifest.xml:
```diff
@@ -6,7 +6,7 @@
   <Identity
     Name="EndlessOSFoundation.EndlessKey.app"
-    Version="0.1.10.1"
+    Version="0.1.11.0"
```
The commit message says:

> the version format is "Major.Minor.Build.Revision",
> where "Major" can never be 0.

but Major here is 0. Don't you need to bump this to e.g. 1.0.0.0 ?

jprvita replied:
Good catch, that completely slip through. I'm surprised the upload validation logic did not catch it. I'll bump it to 1.0.0.0 and update the commit message accordingly.

@danigm FYI, looks like this will app will need to be released as 1.0.0.0 (or greater) once we are ready to publish to the Microsoft Store, to satisfy the store requirements.
